apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/modal.js

```diff
@@ -0,0 +1,65 @@
+import { Button, Modal } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import React from 'react';
+import image from './image.svg';
+
+import './modal.scss';
+
+const GlobalStylesModal = () => {
+	const [ isVisible, setIsVisible ] = useState( true );
+
+	if ( ! isVisible ) {
+		return null;
+	}
+
+	let calypsoDomain = 'https://wordpress.com';
+	const searchParams = new URLSearchParams( window.location.search );
+	const params = Object.fromEntries( searchParams.entries() );
+	const { origin } = params;
+	switch ( origin ) {
+		case 'http://calypso.localhost:3000':
+		case 'https://wpcalypso.wordpress.com':
+		case 'https://horizon.wordpress.com':
+		case 'https://wordpress.com':
+			calypsoDomain = origin;
+			break;
+	}
```

Comment:
This switch with only fall-throughts looks a bit odd to me (even ignoring the fact that the `https://wordpress.com` is redundant 🙂).

How about:
```js
const calypsoDomain = [
	'http://calypso.localhost:3000',
	'https://wpcalypso.wordpress.com',
	'https://horizon.wordpress.com',
].includes( origin )
	? origin
	: 'https://wordpress.com';
```

Aside, this looks like something that someone might have already solved somewhere. I'm aware it's just a few lines, but it might be worth doing a quick search across Calypso, just in case. 🙂

Reply:
> How about:
> ```js
> const calypsoDomain = [
> 	'http://calypso.localhost:3000',
>	'https://wpcalypso.wordpress.com',
>	'https://horizon.wordpress.com',
> ].includes( origin )
>	? origin
>	: 'https://wordpress.com';
> ```

I like it!

> this looks like something that someone might have already solved somewhere.

It's solved in PHP but I couldn't find any similar code in JS: 

https://github.com/Automattic/wp-calypso/blob/09486588ba341fa4b3a949eebb88a430f9f52690/apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php#L87-L97